zh/deploy-on-aws-eks.md

```diff
@@ -5,181 +5,76 @@ summary: 介绍如何在 AWS EKS (Elastic Kubernetes Service) 上部署 TiDB 集
-    ```
-
-    二进制文件下载完成后，执行以下操作：
-
-    {{< copyable "shell-regular" >}}
-
-    ``` shell
-    chmod +x ./aws-iam-authenticator && \
-    sudo mv ./aws-iam-authenticator /usr/local/bin/aws-iam-authenticator
-    ```
-
-## 部署集群
-
-### 部署 EKS、TiDB Operator 和 TiDB 集群节点池
-
-使用如下步骤部署 EKS、TiDB Operator 和 TiDB 集群节点池。
-
-从 Github 克隆代码并进入指定路径：
-
-{{< copyable "shell-regular" >}}
-
-``` shell
-git clone --depth=1 https://github.com/pingcap/tidb-operator && \
-cd tidb-operator/deploy/aws
-```
-
-EKS 部署脚本中提供两种部署方式：
-
-* 部署一套快速体验的 Demo 集群。
-
-    Demo 部署会创建一个新的 VPC、一个 t2.micro 实例作为堡垒机，并包含以下 ec2 实例作为工作节点的 EKS 集群：
-
-    * 1 台 c5.large 实例，部署 PD
-    * 3 台 c5d.large 实例，部署 TiKV
-    * 1 台 c5.large 实例，部署 TiDB
-    * 1 台 c5.large 实例，部署监控组件
-
-    执行如下命令使用 Demo 配置：
-
-    {{< copyable "shell-regular" >}}
-
-    ``` shell
-    cp demo.tfvars terraform.tfvars
-    ```
-
-* 部署一套生产环境集群。
-
-    生产环境集群部署会创建一个新的 VPC、一个 t2.micro 实例作为堡垒机，并包含以下 ec2 实例作为工作节点的 EKS 集群：
-
-    * 3 台 c5.xlarge 实例，部署 PD
-    * 3 台 i3.4xlarge 实例，部署 TiKV
-    * 2 台 c5.2xlarge 实例，部署 TiDB
-    * 1 台 c5.2xlarge 实例，部署监控组件
-
-    执行如下命令使用生产环境配置：
-
-    {{< copyable "shell-regular" >}}
-
-    ``` shell
-    cp prod.tfvars terraform.tfvars
-    ```
-
-除了上述配置，可以编辑 `terraform.tfvars`，在其中设置变量的值，按需配置集群，可以通过 `variables.tf` 查看有哪些变量可以设置以及各变量的详细描述。
-
-例如，下面示例配置 EKS 集群名称，TiDB 集群名称，TiDB Operator 版本及 PD、TiKV 和 TiDB 节点的数量：
-
-```
-default_cluster_pd_count   = 3
-default_cluster_tikv_count = 3
-default_cluster_tidb_count = 2
-default_cluster_name = "tidb"
-eks_name = "my-cluster"
-operator_version = "v1.1.0"
-```
-
-如果需要在集群中部署 TiFlash，需要在 `terraform.tfvars` 中设置 `create_tiflash_node_pool = true`，也可以设置 `cluster_tiflash_count` 和 `cluster_tiflash_instance_type` 来配置 TiFlash 节点池的节点数量和实例类型，`cluster_tiflash_count` 默认为 `2`，`cluster_tiflash_instance_type` 默认为 `i3.4xlarge`。
-
-如果需要在集群中部署 TiCDC，需要在 `terraform.tfvars` 中设置 `create_cdc_node_pool = true`，也可以设置 `cluster_cdc_count` 和 `cluster_cdc_instance_type` 来配置 TiCDC 节点池的节点数量和实例类型，`cluster_cdc_count` 默认为 `3`，`cluster_cdc_instance_type` 默认为 `c5.2xlarge`。
+该教程会指导安装并配置好 AWS 的命令行工具 awscli 以及用来创建 Kubernetes 集群的 eksctl 工具。同时 Kubernetes 命令行 kubectl 也会下载并安装好。
 
 > **注意：**
 >
-> 请通过 `variables.tf` 文件中的 `operator_version` 确认当前版本脚本中默认的 TiDB Operator 版本，如果默认版本不是想要使用的版本，请在 `terraform.tfvars` 中配置 `operator_version`。
-
-配置完成后，使用 `terraform` 命令初始化并部署集群：
+> 本文档操作，需要 AWS Access Key 至少具有 [eksctl 所需最少权限](https://eksctl.io/usage/minimum-iam-policies/) 和创建 [Linux 堡垒机所涉及的服务权限](https://docs.aws.amazon.com/quickstart/latest/linux-bastion/architecture.html#aws-services)。
 
-{{< copyable "shell-regular" >}}
+## 部署
 
-``` shell
-terraform init
-```
+### 创建 EKS 和节点池
 
 {{< copyable "shell-regular" >}}
 
-``` shell
-terraform apply
+```yaml
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+metadata:
+  name: <clusterName>
+  region: us-west-2
+
+nodeGroups:
+  - name: admin
+    desiredCapacity: 1
+    labels:
+      role: admin
```

Comment:
We create a separate node group for the monitor before, with this change the monitor will reuse the `admin` node group?

Reply:
yes, I think it's simpler. It might be useful to create dedicated node groups for tikv, etc. as they may need special hardware, e.g. high-performance CPU/Memory or disks. But for general purpose applications, it's not necessary to create a dedicated node group for each application.